podcasts/Sharing/Clip/VideoExporter.swift

```diff
@@ -191,11 +206,16 @@ fileprivate actor Counter {
 }
 
 fileprivate extension AVAssetWriterInput {
-    func unsafeRequestMediaDataWhenReady(_ block: @escaping (CheckedContinuation<Void, Error>) async -> Void) async throws {
+    func unsafeRequestMediaDataWhenReady(_ block: @escaping () async throws -> Void) async throws {
         try await withCheckedThrowingContinuation { continuation in
             requestMediaDataWhenReady(on: .global(qos: .userInitiated)) {
                 _unsafeWait {
-                    await block(continuation)
+                    do {
```

Comment:
This ensures the continuation is called in every branch and makes the block a bit simpler.